packages/python/plotly/plotly/tests/test_io/test_renderers.py

```diff
@@ -381,3 +382,27 @@ def test_repr_mimebundle_mixed_renderer(fig1):
     assert set(fig1._repr_mimebundle_().keys()) == set(
         {"application/vnd.plotly.v1+json", "text/html"}
     )
+
+
+def test_missing_webbrowser_module(fig1):
+    """
+    Assert that no errors occur if the webbrowser module is absent
+    """
+    removed_webbrowser_module = sys.modules["webbrowser"]
+    del sys.modules["webbrowser"]
```

Comment:
Does this actually cause `import webbrowser` to fail? Let's add something to the test to validate this:
```py
with pytest.raises(ImportError):
    import webbrowser
```

And if (as I think may be the case) it's still importable, here's a solution that looks promising: https://stackoverflow.com/a/60229056/9188800

Reply:
This was helpful in pointing me in the right direction. Thanks! Landed on something similar buy slightly different in the end though.